index.js

```diff
@@ -16,28 +16,84 @@ const collectionName = 'dogs'
 app.get('/dogs', (req, res) => {
     mongoClient.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true }, (err, client) => {
         if (err) {
-            return res.status(500).json({
-                success: false,
-                message: 'Error connecting to database.',
-                data: []
-            })
+            client.close()
+            return composeJSON(res, 500, false, 'Error connecting to database.', [])
         }
         let db = client.db(dbName)
         let collection = db.collection(collectionName)
         collection.find({}).toArray((err, docs) => {
             if (err) {
-                return res.status(500).json({
-                    success: false,
-                    message: 'Error getting data from database.',
-                    data: []
-                })
+                client.close()
+                return composeJSON(res, 500, false, 'Error getting data from database.', [])
+            } else {
+                client.close()
+                return composeJSON(res, 200, true, 'Data retrieved successfully', docs)
             }
-            res.status(200).json({
-                success: true,
-                message: 'Data has been successfully retrieved and sent.',
-                data: docs
+        })
+    })
+})
+
+app.post('/dogs/:id/wins', (req, res) => {
+    const id = req.params.id
+    const regex = RegExp('[0-9a-f]{24}')
+    if (regex.exec(id) === null) {
+        return composeJSON(res, 400, false, 'Invalid winner ID.', [])
+    }
+    mongoClient.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true }, (err, client) => {
+        if (err) {
+            client.close()
+            return composeJSON(res, 500, false, 'Error connecting to database.', [])
+        }
+        const db = client.db(dbName)
+        const collection = db.collection(collectionName)
+        try {
+            declareChampion(collection, id, function (result) {
+                if (result.success === true) {
+                    client.close()
+                    return composeJSON(res, 200, true, 'Dog victory recorded', [result.result.result])
+                } else {
+                    client.close()
+                    return composeJSON(res, 400, false, 'Not able to update records.', [])
+                }
             })
+        }
+        catch (err) {
```

Comment:
`catch` should be on the previous line. Also, as we catch and swallow all errors in the `declareChampion` I don't think that this will ever be hit. So we either need to remove thus block, or allow the errors from `declareChampion` to come through - interested to know what you think

Reply:
done